src/Pos/Block/Error.hs

```diff
@@ -24,11 +26,44 @@ data RollbackException = RollbackTipMismatch HeaderHash HeaderHash
 
 renderRollbackException :: RollbackException -> Text
 renderRollbackException = \case
-    RollbackTipMismatch storedTip attemptedTip->
+    RollbackTipMismatch storedTip attemptedTip ->
         tipMismatchMsg "rollback" storedTip attemptedTip
 
 instance Exception RollbackException where
     displayException = toString . renderRollbackException
 
 instance Buildable RollbackException where
     build = Data.Text.Buildable.build . renderRollbackException
+
+data ApplyBlocksException
+    = ApplyBlocksTipMismatch Text HeaderHash HeaderHash
```

Comment:
Which is expected, which is real? Do i understand correctly that `ApplyBlocksError` captures everything we can't predict? D o c s.